Sources/SKCore/CompilationDatabase.swift

```diff
@@ -99,13 +114,24 @@ public struct FixedCompilationDatabase: CompilationDatabase, Equatable {
 }
 
 extension FixedCompilationDatabase {
-  public init(directory: AbsolutePath, _ fileSystem: FileSystem = localFileSystem) throws {
+  /// Loads the compilation database located in `directory`, if any.
+  /// returns: `nil` if `compile_flags.txt` was not found
+  public init?(directory: AbsolutePath, _ fileSystem: FileSystem = localFileSystem) throws {
     let path = directory.appending(component: "compile_flags.txt")
-    try self.init(file: path, fileSystem)
+    guard let database = try Self(file: path, fileSystem) else {
+      return nil
+    }
+    self = database
```

Comment:
Am I missing something or could this just continue being

```suggestion
    try self.init(file: path, fileSystem)
```

Reply:
Good catch. I've had one foot outside of Swift land the year or so and forgot we can have nice things :)